x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_custom_settings_flyout/crawl_custom_settings_flyout_logic.ts

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../../shared/http';
+import { EngineLogic } from '../../../engine';
+
+import { CrawlerLogic } from '../../crawler_logic';
+import { DomainConfig, DomainConfigFromServer } from '../../types';
+import { domainConfigServerToClient } from '../../utils';
+import { extractDomainAndEntryPointFromUrl } from '../add_domain/utils';
+
+export interface CrawlCustomSettingsFlyoutLogicValues {
+  customEntryPointUrls: string[];
+  customSitemapUrls: string[];
+  domainUrls: string[];
+  domainConfigs: DomainConfig[];
+  domainConfigMap: {
+    [key: string]: DomainConfig;
+  };
+  entryPointUrls: string[];
+  includeRobotsTxt: boolean;
+  isDataLoading: boolean;
+  isFormSubmitting: boolean;
+  isFlyoutVisible: boolean;
+  maxCrawlDepth: number;
+  selectedDomainUrls: string[];
+  selectedEntryPointUrls: string[];
+  selectedSitemapUrls: string[];
+  sitemapUrls: string[];
+}
+
+export interface CrawlCustomSettingsFlyoutLogicActions {
+  fetchDomainConfigData(): void;
+  hideFlyout(): void;
+  onRecieveDomainConfigData(domainConfigs: DomainConfig[]): { domainConfigs: DomainConfig[] };
+  onSelectCustomEntryPointUrls(entryPointUrls: string[]): { entryPointUrls: string[] };
+  onSelectCustomSitemapUrls(sitemapUrls: string[]): { sitemapUrls: string[] };
+  onSelectDomainUrls(domainUrls: string[]): { domainUrls: string[] };
+  onSelectEntryPointUrls(entryPointUrls: string[]): { entryPointUrls: string[] };
+  onSelectMaxCrawlDepth(maxCrawlDepth: number): { maxCrawlDepth: number };
+  onSelectSitemapUrls(sitemapUrls: string[]): { sitemapUrls: string[] };
+  showFlyout(): void;
+  startCustomCrawl(): void;
+  toggleIncludeRobotsTxt(): void;
```

Comment:
The naming here is slightly (semantically) incorrect because the toggle is to include/exclude **sitemaps** from robots.txt.

Reply:
Thanks for clarifying this @orhantoy, I addressed this in https://github.com/elastic/kibana/pull/124999/commits/8d9ca6f193b9f9919c11a082ad888ae456b5c9bc